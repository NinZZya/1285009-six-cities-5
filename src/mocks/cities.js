export default {
  1: {
    id: 1,
    name: `Paris`,
    coords: [48.864716, 2.349014],
  },
  2: {
    id: 2,
    name: `Cologne`,
    coords: [50.9415491, 6.9408247],
  },
  3: {
    id: 3,
    name:
    `Brussels`,
    coords: [50.8503396, 4.3517103],
  },
  4: {
    id: 4,
    name: `Amsterdam`,
    coords: [52.38333, 4.9],
  },
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};
