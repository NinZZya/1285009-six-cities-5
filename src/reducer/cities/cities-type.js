export const SETE_CITIES = `SETE_CITIES`;
export const CHANGE_ACTIVE_CITY_ID = `CHANGE_ACTIVE_CITY_ID`;
