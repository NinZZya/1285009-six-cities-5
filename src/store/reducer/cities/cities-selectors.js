import NameSpace from '~/store/reducer/name-space';


const NAME_SPACE = NameSpace.CITIES;


export const getActiveCityId = (state) => state[NAME_SPACE].activeId;
export const getCities = (state) => state[NAME_SPACE].cities;
