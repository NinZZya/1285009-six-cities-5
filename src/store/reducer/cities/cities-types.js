export const SET_CITIES = `SET_CITIES`;
export const CHANGE_ACTIVE_CITY_ID = `CHANGE_ACTIVE_CITY_ID`;
