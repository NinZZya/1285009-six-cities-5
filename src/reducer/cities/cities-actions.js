import * as CitiesType from './cities-type';
import {makeActionCreator} from '../../utils/utils';


export const updateCities = makeActionCreator(CitiesType.UPDATE_CITIES);
export const changeActiveCityId = makeActionCreator(CitiesType.CHANGE_ACTIVE_CITY_ID);
