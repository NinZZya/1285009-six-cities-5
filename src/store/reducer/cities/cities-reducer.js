import * as CitiesType from './cities-type';
import {extend} from '../../../utils/utils';
import {DEFAULT_CITY_ID} from '../../../constants/const';


const cities = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

const initialState = {
  cities,
  activeId: DEFAULT_CITY_ID,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case CitiesType.SETE_CITIES:
      return extend(state, {
        cities: action.payload,
      });
    case CitiesType.CHANGE_ACTIVE_CITY_ID:
      return extend(state, {
        activeId: action.payload,
      });
    default:
      return state;
  }
};
