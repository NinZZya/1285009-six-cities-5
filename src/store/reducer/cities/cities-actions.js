import * as CitiesType from './cities-types';
import {makeActionCreator} from '../../../utils/utils';


export const setCities = makeActionCreator(CitiesType.SET_CITIES);
export const changeActiveCityId = makeActionCreator(CitiesType.CHANGE_ACTIVE_CITY_ID);
