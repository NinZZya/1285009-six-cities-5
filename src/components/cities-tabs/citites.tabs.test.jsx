import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import CitiesTabs from './cities-tabs';
import history from '../../constants/history';
import cities from '../../mocks/cities';


const ACTIVE_CITY_ID = 1;

it(`Should CitiesTabs render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <CitiesTabs
            activeCityId={ACTIVE_CITY_ID}
            cities={cities}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
