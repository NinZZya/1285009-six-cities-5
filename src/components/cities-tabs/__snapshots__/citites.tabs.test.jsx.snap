// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CitiesTabs render correctly 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          href="/city/1"
          onClick={[Function]}
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="/city/2"
          onClick={[Function]}
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="/city/3"
          onClick={[Function]}
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="/city/4"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="/city/5"
          onClick={[Function]}
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="/city/6"
          onClick={[Function]}
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
